python/ray/tune/suggest/optuna.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import pickle
+
+from ray.tune.result import TRAINING_ITERATION
+
+try:
+    hyperopt_logger = logging.getLogger("hyperopt")
+    hyperopt_logger.setLevel(logging.WARNING)
+    import optuna as ot
+except ImportError:
+    ot = None
+
+from ray.tune.suggest import Searcher
+
+logger = logging.getLogger(__name__)
+
+
+class _Param:
+    def __getattr__(self, item):
+        def _inner(*args, **kwargs):
+            return (item, args, kwargs)
+
+        return _inner
+
+
+param = _Param()
+
```

Comment:
Alternatively, we could explicitly define the `suggest_` functions (so that code completion works in IDEs).

Also, these could then be class methods of `OptunaSearch` - so something like `OptunaSearch.suggest_categorical()` would be an alternative API.